virtualmachine/vsphere/vm.go

```diff
@@ -1678,49 +1690,104 @@ func CreateTemplate(vm *VM) error {
 	return err
 }
 
-// GetTemplateList : Returns the template VMs in a cluster
-func GetTemplateList(vm *VM) ([]map[string]interface{}, error) {
+// getOsDetails: returns details of guest os
+func getOsDetails(vmMo mo.VirtualMachine) map[string]interface{} {
+	osDetails := make(map[string]interface{})
+	if vmMo.Guest == nil {
+		return osDetails
+	}
+	osDetails = map[string]interface{}{
+		"guest_id":        vmMo.Guest.GuestId,
+		"guest_family":    vmMo.Guest.GuestFamily,
+		"guest_full_name": vmMo.Guest.GuestFullName,
+		"hostname":        vmMo.Guest.HostName,
+	}
+	return osDetails
+}
+
+// getVmInfo : get vm info from vm managed object
+func getVmInfo(vmMo mo.VirtualMachine, vmPath string) map[string]interface{} {
+	var (
+		toolsStatus bool
+	)
+	// fetching fisk info
+	diskInfo := make([]map[string]interface{}, 0)
+	for _, device := range vmMo.Config.Hardware.Device {
+		disk, ok := device.(*types.VirtualDisk)
+		if !ok {
+			continue
+		}
+		devinfo, ok := disk.DeviceInfo.(*types.Description)
+		if !ok {
+			continue
+		}
+		backing := disk.Backing.(types.BaseVirtualDeviceFileBackingInfo)
+		fileBackingInfo := backing.GetVirtualDeviceFileBackingInfo()
+		diskInfo = append(diskInfo, map[string]interface{}{
+			"name":      devinfo.Label,
+			"size":      disk.CapacityInKB,
+			"disk_file": fileBackingInfo.FileName,
+		})
+	}
+	if vmMo.Guest != nil {
```

Comment:
Missing error handling for (vmMo.Guest == nil) case?

Reply:
in case guest is nil(when the vm is not ready yet), toolStatus will be `false` which is the desired status.